craft_application/errors.py

```diff
@@ -141,16 +217,43 @@ def __init__(self, host_directive: str) -> None:
         super().__init__(message=message)
 
 
-class InvalidPlatformError(CraftError):
+class PlatformDefinitionError(CraftError):
+    """Errors with the platform definitions."""
+
+
+class InvalidPlatformError(PlatformDefinitionError):
     """The selected build plan platform is invalid."""
 
     def __init__(self, platform: str, all_platforms: Sequence[str]) -> None:
         message = f"Platform {platform!r} not found in the project definition."
-        details = f"Valid platforms are: {', '.join(all_platforms)}."
+        platforms_str = ", ".join(repr(platform) for platform in all_platforms)
+        details = f"Valid platforms are: {platforms_str}."
 
         super().__init__(message=message, details=details)
 
 
+class ArchitectureNotInPlatformError(PlatformDefinitionError):
+    """The selected build-for is not in the selected platform."""
```

Comment:
It seems that it handles `build-on` as well
```suggestion
    """The selected architectures is not a part of the selected platform."""
```